### PR TITLE
Add demo products importer

### PR DESCRIPTION
**Pull request by am-on**
Refs: https://github.com/niteoweb/woocart/issues/322

### CODE REVIEW

**Comment by am-on** [comment]
imported product on store:
<details>

image

</details>

File: tests/WooProductTest.php [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+    /**
+     * @covers \Niteo\WooCart\Defaults\Importers\WooProducts::absolute_image_path
+     * @covers \Niteo\WooCart\Defaults\Importers\WooProducts::add_products
+     * @covers \Niteo\WooCart\Defaults\Importers\WooProducts::create_simple_product
+     * @covers \Niteo\WooCart\Defaults\Importers\WooProducts::get_product_count
+     * @covers \Niteo\WooCart\Defaults\Importers\WooProducts::parse_product
+     * @covers \Niteo\WooCart\Defaults\Importers\WooProducts::upload_images
+     */
+    function testadd_products() {
+        $mock_WC_Product = \Mockery::mock('overload:WC_Product');
```
Comment (dz0ny) [inline tests/WooProductTest.php]: Assert WC_Product is called three times...

File: src/importers/class-wooproducts.php [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+		 * Read file, parse and add products.
+		 *
+		 * @param string $file_path
+		 */
+		public function add_products( $file_path ) {
+			$this->file_path = $file_path;
+			$this->base_path = dirname( $file_path );
+
+			$contents = file_get_contents( $this->file_path );
+			$products = explode('---', $contents);
```
Comment (dz0ny) [inline src/importers/class-wooproducts.php]: regexp ^---$

Because --- can be used in text description

File: tests/WooProductTest.php [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+        $faker->ean8 = 45321150;
+        $mock_FakerFactory = \Mockery::mock('overload:Faker\Factory');
+        $mock_FakerFactory->shouldReceive('create')->andReturn($faker);
+        \WP_Mock::wpFunction(
+            'sanitize_title', array(
+                'return'    => 'product-title'
+            )
+        );
+
+        $data = array(
```
Comment (dz0ny) [inline tests/WooProductTest.php]: Product meta should be a class https://github.com/woocart/woocart-defaults/blob/master/src/importers/class-woopage.php#L24

File: src/importers/class-wooproducts.php [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+					'featured'           => $faker->boolean(),
+					'catalog_visibility' => 'visible',
+					'description'        => $data['description'] . $data['details'],
+					'short_description'  => null,
+					'sku'                => sanitize_title( $name ) . '-' . $faker->ean8,
+					'regular_price'      => $price,
+					'sale_price'         => $sale_price,
+					'date_on_sale_from'  => '',
+					'date_on_sale_to'    => $faker->iso8601( date( 'c', strtotime( '+1 month' ) ) ),
+					'total_sales'        => $faker->numberBetween( 0, 10000 ),
```
Comment (dz0ny) [inline src/importers/class-wooproducts.php]: 0

File: src/importers/class-wooproducts.php [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+		}
+
+
+		/**
+		 * Transform image path from relative to absolute.
+		 * @param string $image
+		 * @param string $replacement
+		 * @return array
+		 */
+		private function absolute_image_path( $image_path, $replacement='common:' ) {
```
Comment (dz0ny) [inline src/importers/class-wooproducts.php]: common: is alias for "/provision/localizations/Countries/.common/"

File: src/classes/class-cli-command.php [hunk 6]
```diff
@@ -163,5 +164,32 @@ public function products( $args, $assoc_args ) {
...
 		}
 
+		/**
+		 * Import demo products.
+		 *
+		 * ## OPTIONS
+		 *
+		 * <path>
+		 * : Path to file with products
+		 *
```
Comment (dz0ny) [inline src/classes/class-cli-command.php]:
```
[--common_path]
: Path to .common directory for localizations
---
default: /provision/localizations/Countries/.common/
---
```
see https://make.wordpress.org/cli/handbook/commands-cookbook/#annotating-with-phpdoc

**Comment by dz0ny** [comment]
Also, change to version v3.2.0 in makefile.

**Pull request merged**